doc/nrf/ug_radio_coex.rst

```diff
@@ -0,0 +1,111 @@
+.. _ug_radio_coex:
+
+Working with WiFi Coexistence in Short Range Radio applications
+###############################################################
+
+.. contents::
+   :local:
+   :depth: 4
+
+Short Range RF technologies such as BLE or 802.15.4 (from now on refered to as SR) use different radios than WiFi. However, if both SR and WiFi
+attempt to transmit at the same time RF waves will interfere with each other leading to decreased performance and uneeded higher power
+consumption. Also, in some cases (such as receiving ACK) no radio should be transmitting in order to ensure correct reception.
+
+In order to mitigate coexistence issues and improve coexistence performance a Packet Traffic Arbiter (PTA) is used. Both SR and WiFi request access to RF
+and PTA grants/denies that access.
+
+This guide describes how to add WiFi Coexistence (CX) support to your SR application in |NCS|.
+
+
+Software support
+****************
+
+If your application uses SR protocols and requires coexistence with WiFi, you must enable :ref:`nrfxlib:mpsl_cx` in the
+:ref:`nrfxlib:mpsl` (MPSL) library.
+
+CX with MPSL
+============
+
+The MPSL library provides an implementation for the WiFi Coexistens module for SR applications.
+To use this implementation, your application must use a protocol driver that enables the WiFi Coexistence (CX) feature.
+The library provides multiprotocol support, but you can also use it in applications that require only one protocol.
+To avoid conflicts, check the protocol documentation to see if the protocol uses CX support provided by MPSL.
+
+.. note::
+   Currently, the following protocols use CX support provided by MPSL:
+
+   * :ref:`ug_thread`
+   * :ref:`ug_zigbee`
+   * :ref:`ug_ble_controller`
+   * :ref:`ug_multiprotocol_support`
+
+.. _ug_radio_cx_requirements:
+
+Enabling MPSL
+-------------
+
+Before you add the devicetree node in your application, complete the following steps:
+
+1. Add support for the MPSL library in your application.
+   The MPSL library provides API to handle signals coming from PTA and controlling requests send to PTA.
+   See :ref:`nrfxlib:mpsl_lib` in the nrfxlib documentation for details.
+2. Enable support for MPSL implementation in |NCS| by setting the :kconfig:`CONFIG_MPSL` Kconfig option to ``y``.
+
+Enabling CX
+-------------
+
+In order to enable CX set the :kconfig:`CONFIG_MPSL_CX` Kconfig option to ``y``.
+
+Hardware description
```

Comment:
I like this distinction you're introducing here. In order to enable WiFi Coexistence, a user must specify:
* details of the hardware interface used to communicate with PTA, and that information is stored in devicetree;
* details of the software to be used for controlling that interface, and that information is stored in Kconfig.

I think we can show that distinction in the doc - it helps understand what needs to be done and why.

Moreover, in case of nRF53 all that information must be provided _both_ to the nRF53 application core and to the nRF53 network core. I see that there's a section below that explains nicely how to enable support for the generic 3-pin interface - maybe we could try to make that section generic. Probably most of the things mentioned in that section ("add a node to the DTS file", "pass flags to the child image") apply to all interfaces. The generic 3-pin interface could be used as an example, "developed" alongside the generic description, to show step by step what needs to be done in this simple case

Reply:
Well all the steps when it comes to hardware description for both the "thread" and "generic three pin interface" implementations are identical, so now I've moved this steps to the top level of the section and emphasized this is how it work for all currently supported implementations, but can be changed in a custom user implementation.